node/src/proxy_client/stream_handler_pool.rs

```diff
@@ -1735,6 +1919,49 @@ mod tests {
         );
     }
 
+    #[test]
+    fn clean_up_dead_streams_logs_when_the_shutdown_channel_is_down() {
```

Comment:
Maybe a little higher-level test name? Something about the `StreamReader` being down, rather than a low-level reference to the shutdown channel?